service/kvmconfig/v3/resource/clusterrolebinding/current.go

```diff
@@ -0,0 +1,52 @@
+package clusterrolebinding
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	apiv1 "k8s.io/api/rbac/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	apismetav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/giantswarm/kvm-operator/service/kvmconfig/v3/key"
+)
+
+func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+	r.logger.LogCtx(ctx, "debug", "looking for a list of cluster role bindings in the Kubernetes API")
+
+	var currentClusterRoleBinding []*apiv1.ClusterRoleBinding
+	{
+		clusterRoleBinding, err := r.k8sClient.RbacV1beta1().ClusterRoleBindings().Get(key.ClusterRoleBindingName(customObject), apismetav1.GetOptions{})
+		if apierrors.IsNotFound(err) {
+			r.logger.LogCtx(ctx, "debug", "did not find cluster role binding in the Kubernetes API")
+			// fall through
+		} else if err != nil {
+			return nil, microerror.Mask(err)
+		} else {
+			r.logger.LogCtx(ctx, "debug", "found a list of cluster role binding in the Kubernetes API")
+
+			currentClusterRoleBinding = append(currentClusterRoleBinding, clusterRoleBinding)
+		}
+
+		clusterRoleBindingPSP, err := r.k8sClient.RbacV1beta1().ClusterRoleBindings().Get(key.ClusterRoleBindingPSPName(customObject), apismetav1.GetOptions{})
```

Comment:
What is the difference between a role binding and a role binding PSP? Why do we need both? Why is both handled by the same resource?

Reply:
The reason why I have made a different binding for Pod Security Policies is granularity. It is debatable, but I have found clearer have a different binding for psp (borrowed from some community helm charts). Because `Pod Security Policies` is a special resource, I did not want to set it in the same rule list than other resources. It has more sense in the [static files](https://github.com/giantswarm/kvm-operator/blob/master/helm/kvm-operator-chart/templates/rbac.yaml) because improves readability and separation of concerns.